sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs

```diff
@@ -131,9 +104,66 @@ internal static void CastAndRemoveAnnotations(this IDictionary<string, object> e
             }
         }
 
-        private static string ToOdataTypeString(this string name)
+        /// <summary>
+        /// Converts a List of Dictionaries containing properties and Odata type annotations to a custom entity type.
+        /// </summary>
+        internal static List<T> ToTableEntityList<T>(this IReadOnlyList<IDictionary<string, object>> entityList) where T : TableEntity, new()
         {
-            return $"{name}{TableConstants.Odata.OdataTypeString}";
+            var properties = typeof(T).GetProperties(BindingFlags.Instance | BindingFlags.Public);
```

Comment:
We would eventually need to do something with perf..

Reply:
This was intended to be a minimally invasive to generated code implementation. We could certainly implement custom serialization at some point. It would be great to chat about any implementation approaches you have in mind,